frontend/packages/dev-console/src/components/project-access/hooks.ts

```diff
@@ -0,0 +1,41 @@
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { ClusterRoleModel } from '@console/internal/models';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+export type ClusterRoleKind = K8sResourceCommon & {
+  rules: { [key: string]: string[] }[];
+};
+
+export enum accessRoles {
+  admin = 'Admin',
+  edit = 'Edit',
+  view = 'View',
+}
+// for testing until backend support
+// eslint-disable-next-line dot-notation
+window.SERVER_FLAGS['availableClusterRoles'] = ['admin', 'view', 'edit', 'cluster-reader'];
+export const useProjectAccessRoles = () => {
+  // eslint-disable-next-line dot-notation
+  const availableClustersRoles = window.SERVER_FLAGS['availableClusterRoles'];
+  const [clusterRoles, clusterRolesLoaded] = useK8sWatchResource<ClusterRoleKind[]>({
+    kind: ClusterRoleModel.kind,
+    isList: true,
+  });
```

Comment:
Can we check the `availableClusterRoles` here to fetch the ClusterRoles only if necessary?

```suggestion
  const [clusterRoles, clusterRolesLoaded] = useK8sWatchResource<ClusterRoleKind[]>(availableClustersRoles ? {
    kind: ClusterRoleModel.kind,
    isList: true,
  } : {});
```

Reply:
We can not pass an empty object to `useK8sWatchResource` as kind is the required field.